spec/routemaster/integration/api_client_spec.rb

```diff
@@ -242,49 +221,57 @@ class ProcHandler
     it 'does not make any http requests to fetch resources any if just the index method is called' do
       resources = subject.discover(url).resources
 
-      expect(subject).to receive(:get).with("http://localhost:8000/resources", anything).once
+      expect(subject).to receive(:get).with("http://localhost:#{port}/resources", anything).once
       resources.index
     end
   end
 
   describe 'Telemetry' do
-    let(:metrics_client) { double('MetricsClient') }
+    let(:metrics_client) { Dogstatsd.new }
     let(:source_peer) { 'test_service' }
     let(:url) { "#{host}/success" }
+    let(:calls) { [] } 
 
     subject do
       Routemaster::APIClient.new(metrics_client: metrics_client,
                                  source_peer: source_peer)
     end
 
+    before do
+      wrapper = -> (m, *args, &block) {
+        calls << [m.name, *args]
+        m.call(*args, &block)
+      }
```

Comment:
As I understand it this is a spy right? https://relishapp.com/rspec/rspec-mocks/docs/basics/spies